src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/TornadoService.java

```diff
@@ -217,4 +232,61 @@ private void writeOutputStream(OutputStreamWriter outputStreamWriter, StringBuil
         outputStreamWriter.write(sb.toString());
         outputStreamWriter.flush();
     }
+
+    /**
+     * This method calls the helper method to create the data to be passed through to Tornado and then checks whether
+     * it can reach the service.
+     * @param caseData contains the data needed to generate the PDF
+     * @param userToken contains the user authentication token
+     * @param caseTypeId reference for which casetype the document is being uploaded to
+     * @param documentName name of the document
+     * @return DocumentInfo which contains the URL and markup of the uploaded document
+     * @throws IOException if the call to Tornado has failed, an exception will be thrown. This could be due to
+     timeout or maybe a bad gateway.
+     */
+    public DocumentInfo generateEventDocument(CaseData caseData, String userToken, String caseTypeId,
+                                              String documentName)
+        throws IOException {
+        HttpURLConnection connection = null;
+        try {
+            connection = createConnection();
+            buildDocumentInstruction(connection, caseData, documentName, caseTypeId);
+            return checkResponseStatus(userToken, connection, documentName, caseTypeId);
+        } catch (IOException exception) {
+            log.error(UNABLE_TO_CONNECT_TO_DOCMOSIS, exception);
+            throw exception;
+        } finally {
+            closeConnection(connection);
+        }
+    }
+
+    private void buildDocumentInstruction(HttpURLConnection connection, CaseData caseData, String documentName,
+                                          String caseTypeId)
+            throws IOException {
+        String documentContent = getDocumentContent(caseData, documentName, caseTypeId);
+
+        try (OutputStreamWriter outputStreamWriter = new OutputStreamWriter(connection.getOutputStream(),
+                StandardCharsets.UTF_8)) {
+            outputStreamWriter.write(documentContent);
+            outputStreamWriter.flush();
+        }
+    }
+
+    private String getDocumentContent(CaseData caseData, String documentName, String caseTypeId)
+            throws JsonProcessingException {
+        if (isNullOrEmpty(documentName)) {
+            throw new NullPointerException("Document name cannot be null or empty");
+        }
+        switch (documentName) {
+            case "ET1 Vetting.pdf":
+                return Et1VettingHelper.getDocumentRequest(caseData, tornadoConnection.getAccessKey());
+            case "ET3 Processing.pdf":
+                return Et3VettingHelper.getDocumentRequest(caseData, tornadoConnection.getAccessKey());
+            case "Initial Consideration.pdf" :
+                return InitialConsiderationHelper.getDocumentRequest(
+                        caseData, tornadoConnection.getAccessKey(), caseTypeId);
+            default:
+                throw new IllegalArgumentException("Unexpected document name " + documentName);
+        }
+    }
 }
```

Comment:
Minor point: Consider only one exit from the method as it confirms to usual expectation.

Reply:
Saw this late but as mentioned in messages, it's too late to be changing production ready code